azurerm/resource_arm_application_gateway.go

```diff
@@ -408,18 +407,28 @@ func resourceArmApplicationGateway() *schema.Resource {
 						"backend_address_pool_name": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ConflictsWith: []string{"redirect_configuration_name"},
```

Comment:
Unfortunately as these are nested conflicts with won't work. The only way to actually validate this is upon expand check for the conflict and bubble up an error if they are both set.

Reply:
ok willdo

Reply:
Replaced with validation as suggested